apps/aeutil/lib/list.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Aeutil.List do
+  @moduledoc """
+  This module holds some additional utility functions for list
+  """
+
+  @doc """
+  Merge two list in descending order
+  """
+  @spec merge_descending(list(), list()) :: list()
+  def merge_descending(list_1, list_2) do
+    merge_descending(list_1, list_2, [])
+  end
+
+  @spec merge_descending(list(), list(), list()) :: list()
+  defp merge_descending([], [], acc) do
+    acc
+    |> Enum.sort()
+    |> Enum.reverse()
+  end
+
+  defp merge_descending([], [head2 | list_2], acc) do
+    merge_descending([], list_2, [head2 | acc])
+  end
+
+  defp merge_descending([head1 | list_1], list_2, acc) do
+    case Enum.member?(list_2, head1) do
```

Comment:
If I do it this way, we lose the generic part of the func, because the head in our case is a map `%{height: height, hash: header_hash}`. Yes, I could match it, but then no other two list will match this scenario. So maybe I should move it to the `Aecore.Sync.Chain` module again, if I'll do it this way.

Reply:
But `Enum.sort` that is currently used also uses the `<` ordering (I think), so there should be no difference between the current code and my proposal.